networkx/generators/tests/test_classic.py

```diff
@@ -355,30 +358,85 @@ def test_lollipop_graph_size_node_sequence(self, m, n):
+            ("", -1, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("", 0, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("", "", pytest.raises(nx.NetworkXError)),  # m < 2
+            ("", 1, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("", "a", pytest.raises(nx.NetworkXError)),  # m < 2
+            (1, -1, pytest.raises(nx.NetworkXError)),  # m < 2
+            (1, 0, pytest.raises(nx.NetworkXError)),  # m < 2
+            (1, "", pytest.raises(nx.NetworkXError)),  # m < 2
+            (1, 1, pytest.raises(nx.NetworkXError)),  # m < 2
+            (1, "a", pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", -1, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", 0, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", "", pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", 1, pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", "a", pytest.raises(nx.NetworkXError)),  # m < 2
+            ("a", "b", pytest.raises(nx.NetworkXError)),  # m < 2
+            (2, -1, pytest.raises(nx.NetworkXError)),  # n < 0
+            (2, 0, does_not_raise()),  # ok
+            (2, "", does_not_raise()),  # ok
+            (2, 1, does_not_raise()),  # ok
+            (2, "a", does_not_raise()),  # ok
+            (2, "ab", does_not_raise()),  # ok
+            (2, "aa", pytest.raises(nx.NetworkXError)),  # duplicate node
+            ("ab", -1, pytest.raises(nx.NetworkXError)),  # n < 0
+            ("ab", 0, does_not_raise()),  # ok
+            ("ab", "", does_not_raise()),  # ok
+            ("ab", 1, does_not_raise()),  # ok
+            ("ab", "c", does_not_raise()),  # ok
+            ("ab", "a", pytest.raises(nx.NetworkXError)),  # duplicate node
+            ("aa", "b", pytest.raises(nx.NetworkXError)),  # duplicate node
+            ("aa", "bb", pytest.raises(nx.NetworkXError)),  # duplicate node
+        ],
+    )
+    def test_lollipop_graph_exceptions_m_n(self, m, n, expectation):
+        with expectation:
+            nx.lollipop_graph(m, n)
 
-        # raise NetworkXError if create_using is directed
-        with pytest.raises(nx.NetworkXError):
-            nx.lollipop_graph(2, 20, create_using=nx.DiGraph)
-        with pytest.raises(nx.NetworkXError):
-            nx.lollipop_graph(2, 20, create_using=nx.MultiDiGraph)
+    @pytest.mark.parametrize(
+        ("create_using", "expectation"),
+        [
+            (nx.Graph, does_not_raise()),  # ok
+            (nx.Graph([(0, 1)]), does_not_raise()),  # ok
+            (nx.DiGraph, pytest.raises(nx.NetworkXError)),  # directed
+            (nx.DiGraph([(0, 1)]), pytest.raises(nx.NetworkXError)),  # directed
+            (nx.MultiGraph, does_not_raise()),  # ok
+            (nx.MultiGraph([(0, 1)]), does_not_raise()),  # ok
+            (nx.MultiDiGraph, pytest.raises(nx.NetworkXError)),  # directed
+            (nx.MultiDiGraph([(0, 1)]), pytest.raises(nx.NetworkXError)),  # directed
```

Comment:
I see. In my personal opinion, it doesn't make sense to allow something but discourage it (implictly? I didn't see it mentioned anywhere, but I might be missing something). I know this would _majorly_ suck to deprecate everywhere `create_using` is used, but I can't come up with another "good" reason to handle things this way.